addons/slideshow/functions/fnc_moduleInit.sqf

```diff
@@ -0,0 +1,39 @@
+/*
+ * Author: Jonpas
+ * Initializes the module.
+ *
+ * Arguments:
+ * 0: The module logic <LOGIC>
+ * 1: Units <ARRAY>
+ * 2: Activated <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Public: No
+ */
+//#define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+if !(isServer) exitWith {};
+
+PARAMS_3(_logic,_units,_activated);
+
+if !(_activated) exitWith {};
+
+private ["_objects", "_controllers", "_images", "_names", "_duration"];
+
+_logic = [_this, 0, objNull, [objNull]] call BIS_fnc_param;
+if (isNull _logic) exitWith {};
+
+// Extract variables from logic
+_objects = [_logic getVariable ["Objects", ""], true, true] call FUNC(makeList);
+_controllers = [_logic getVariable ["Controllers", ""], true, true] call FUNC(makeList);
+_images = [_logic getVariable ["Images", ""], true, false] call FUNC(makeList);
+_names = [_logic getVariable ["Names", ""], true, false] call FUNC(makeList);
+_duration = _logic getVariable ["Duration", ""];
+
+// Prepare with actions
+[_objects, _controllers, _images, _names, _duration] call FUNC(createSlideshow);
```

Comment:
Objects and Controllers can be synced objects (addde to the array in `createSlideshow` function), images and names require input indeed.